src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -0,0 +1,262 @@
+import * as React from "react";
+import {Gene, MolecularProfile} from "../../../shared/api/generated/CBioPortalAPI";
+import {action, autorun, computed, IReactionDisposer, observable, ObservableMap} from "mobx";
+import {observer, Observer} from "mobx-react";
+import {AlterationTypeConstants, MutationCache, NumericGeneMolecularDataCache} from "../ResultsViewPageStore";
+import Select from "react-select";
+import DefaultTooltip from "../../../shared/components/defaultTooltip/DefaultTooltip";
+import {remoteData} from "../../../shared/api/remoteData";
+import internalClient from "../../../shared/api/cbioportalInternalClientInstance";
+import {MobxPromise} from "mobxpromise";
+import {CoExpression, CoExpressionFilter} from "../../../shared/api/generated/CBioPortalAPIInternal";
+import _ from "lodash";
+import {IDataQueryFilter} from "../../../shared/lib/StoreUtils";
+import {MSKTab, MSKTabs} from "../../../shared/components/MSKTabs/MSKTabs";
+import CoExpressionViz from "./CoExpressionViz";
+import GeneMolecularDataCache from "../../../shared/cache/GeneMolecularDataCache";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import MutationDataCache from "../../../shared/cache/MutationDataCache";
+import InfoIcon from "../../../shared/components/InfoIcon";
+import {filterAndSortProfiles} from "./CoExpressionTabUtils";
+import MobxPromiseCache from "../../../shared/lib/MobxPromiseCache";
+import {onMobxPromise} from "../../../shared/lib/onMobxPromise";
+
+export interface ICoExpressionTabProps {
+    molecularProfiles:MolecularProfile[];
+    genes:Gene[];
+    studyToDataQueryFilter:{[studyId:string]:IDataQueryFilter}
+    numericGeneMolecularDataCache:NumericGeneMolecularDataCache;
+    mutationCache:MutationCache;
+}
+
+export type CoExpressionCache = MobxPromiseCache<{entrezGeneId:number, molecularProfile:MolecularProfile}, CoExpression[]>;
+
+@observer
+export default class CoExpressionTab extends React.Component<ICoExpressionTabProps, {}> {
+    @observable selectedMolecularProfile:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded, at which point its set automatically (see componentWillMount) and cant be made undefined again
+    @observable selectedEntrezGeneId:number | undefined; // only undefined initially, until genes downloaded, at which point its set automatically (see componentWillMount) and cant be made undefined again
+
+    private allDataRequested:ObservableMap<boolean>;// `${entrezGeneId},${molecularProfileId}`-> boolean
+    private highlightedCoExpression:ObservableMap<CoExpression>;// `${entrezGeneId},${molecularProfileId}`-> CoExpression
+
+    constructor(props:ICoExpressionTabProps) {
+        super(props);
+
+        (window as any).resultsViewCoExpressionTab = this; // for testing
```

Comment:
is this okay for production?

Reply:
this is what @alisman has recommended (I agree) as the least evil way to expose this for e2e testing purposes

Reply:
fair enough, as long as we don't accidentally overwrite some existing global variable this way :)

Reply:
I think thats a good point and I was going to do something about that, I'll do that for this PR

Reply:
not too worried about this.  maybe we namespace these so that there's only ever one thing directly on the global.

Reply:
pushed